File: examples/kotlin/core/shared/app_infra_contracts.kt
package app.protocol.examples.kotlin.core.shared

interface AppHttpClient {
    suspend fun request(config: Any?): Any?
}

interface AppStorageClient {
    suspend fun get(key: String): Any?

    suspend fun set(key: String, value: Any?): Unit
}

interface AppCache {
    suspend fun get(key: String): Any?

    suspend fun set(key: String, value: Any?, ttl: Int? = null): Unit
}

interface AppEventPublisher {
    suspend fun publish(event: String, payload: Any?): Unit
}

File: examples/flutter/core/shared/app_infra_contracts.dart
abstract interface class AppHttpClient {
  Future<dynamic> request(dynamic config);
}

abstract interface class AppStorageClient {
  Future<dynamic> get(String key);

  Future<void> set(String key, dynamic value);
}

abstract interface class AppCache {
  Future<dynamic> get(String key);

  Future<void> set(String key, dynamic value, [int? ttl]);
}

abstract interface class AppEventPublisher {
  Future<void> publish(String event, dynamic payload);
}
